metsim/disaggregate.py

```diff
@@ -7,107 +7,179 @@
+
+    df_disagg['precip'] = precip(df_daily['precip'], ts)
+    df_disagg['wind'] = wind(df_daily['wind'], ts)
+    df_disagg['pet'] = df_daily['pet'].resample(str(int(ts))+'T').fillna(method='ffill')
+
     return df_disagg.fillna(method='ffill')
 
 
-def set_min_max_hour(disagg_rad, n_days):
+def set_min_max_hour(disagg_rad, n_days, ts):
     """
     TODO
     """
-    ts = float(params['time_step'])
     rad_mask = 1*(disagg_rad > 0)
     diff_mask = np.diff(rad_mask)
     rise_times = np.where(diff_mask>0)[0] * ts 
     set_times = np.where(diff_mask<0)[0] * ts
     t_Tmax = 0.67 * (set_times - rise_times) + rise_times
-    t_Tmin = rise_times - float(params['time_step'])
+    t_Tmin = rise_times 
     return t_Tmin, t_Tmax
 
 
-def temp(df_daily, df_disagg):
+def temp(df_daily, df_disagg, t_Tmin, t_Tmax, ts):
     """
     TODO
     """
     # Calculate times of min/max temps
-    n_days = len(df_daily['day_of_year'])
-    t_Tmin, t_Tmax = set_min_max_hour(df_disagg['shortwave'], n_days)
     time = np.array(list(next(it) for it in itertools.cycle(
-                [iter(t_Tmin), iter(t_Tmax)])))
+                [iter(t_Tmin), iter(t_Tmax)])))  
     temp = np.array(list(next(it) for it in itertools.cycle(
                 [iter(df_daily['t_min']), iter(df_daily['t_max'])])))
+    # Account for end points
+    ts_ends = cnst.MIN_PER_HOUR * cnst.HOURS_PER_DAY
+    time = np.append(np.insert(time, 0, time[0:2]-ts_ends), time[-2:]+ts_ends)
+    temp = np.append(np.insert(temp, 0, temp[0:2]), temp[-2:])
     
+    # Interpolate the values
     try:
         interp = scipy.interpolate.PchipInterpolator(time, temp, extrapolate=True)
-        temps = interp(float(params['time_step']) * np.arange(0, len(df_disagg.index)))
+        temps = interp(ts * np.arange(0, len(df_disagg.index)))
     except ValueError:
         temps = np.full(len(df_disagg.index), np.nan)
 
     return temps
 
 
-def precip(precip):
+def precip(precip, ts):
     """
     Splits the daily precipitation evenly throughout the day
     """
-    scale = int(params['time_step']) / (consts['MIN_PER_HOUR'] * consts['HOURS_PER_DAY'])
-    return (precip*scale).resample(params['time_step']+'T').fillna(method='ffill')
+    scale = int(ts) / (cnst.MIN_PER_HOUR * cnst.HOURS_PER_DAY)
+    return (precip*scale).resample(str(int(ts))+'T').fillna(method='ffill')
 
 
-def longwave(longwave):
+def wind(wind, ts):
     """
-    Splits the daily longwave evenly throughout the day
+    Wind is assumed constant throughout the day
     """
-    return longwave.resample(params['time_step']+'T').fillna(method='ffill')
+    return wind.resample(str(int(ts))+'T').fillna(method='ffill')
 
 
-def wind(wind):
+def relative_humidity(vapor_pressure, temp):
     """
-    Wind is assumed constant throughout the day
+    TODO
     """
-    return wind.resample(params['time_step']+'T').fillna(method='ffill')
-
+    return 100. * np.minimum(vapor_pressure*1000/svp(temp), 1.0)
```

Comment:
I'm having a hard time thinking of a way that switching the `1000` our for a constant would improve the code.  Naming it `THOUSAND` seems really pedantic, and we can't just call it `CONVERSION_FACTOR` since that's not descriptive enough.  Doing something like `KILO` could work, but realistically that's exactly the same as putting `THOUSAND`.  What would you suggest putting for these?